e2e/tests/common/mod.rs

```diff
@@ -0,0 +1,66 @@
+use postgres::Client;
+use std::env;
+use std::process::{Command, Stdio};
+use std::str::from_utf8;
+use testcontainers::clients::Cli;
+use testcontainers::images::generic::{GenericImage, WaitFor};
+use testcontainers::{images, Container, Docker};
+
+pub const DB: &str = "postgres-db-test";
+pub const USER: &str = "postgres-user-test";
+pub const PASSWORD: &str = "postgres-password-test";
+
+pub fn run_postgres(client: &Cli) -> Container<Cli, GenericImage> {
+    let docker_image = env::var("TS_DOCKER_IMAGE").unwrap_or(String::from(
+        "ghcr.io/timescale/dev_promscale_extension:develop-ts2-pg14",
+    ));
+
+    let src = concat!(env!("CARGO_MANIFEST_DIR"), "/testdata");
+    let dest = "/testdata";
+
+    let generic_postgres = images::generic::GenericImage::new(docker_image)
+        .with_wait_for(WaitFor::message_on_stderr(
+            "database system is ready to accept connections",
+        ))
+        .with_env_var("POSTGRES_DB", DB)
+        .with_env_var("POSTGRES_USER", USER)
+        .with_env_var("POSTGRES_PASSWORD", PASSWORD)
+        .with_volume(src, dest);
+
+    client.run(generic_postgres)
+}
+
+#[allow(dead_code)]
+pub fn connect(node: &Container<Cli, GenericImage>) -> Client {
+    let connection_string = &format!(
+        "postgres://{}:{}@localhost:{}/{}",
+        USER,
+        PASSWORD,
+        node.get_host_port(5432).unwrap(),
+        DB
+    );
+
+    postgres::Client::connect(connection_string, postgres::NoTls).unwrap()
+}
+
+#[allow(dead_code)]
+pub fn exec_query(node: &Container<Cli, GenericImage>, query_path: &str) -> String {
+    let id = node.id();
+    let abs_query_path = "/".to_owned() + query_path;
+    let output = Command::new("docker")
```

Comment:
It relies on `docker` command being available on the system, where the test runs, doesn't it?  I think this needs to be mentioned in the readme.